pkg/text/template/manual.go

```diff
@@ -15,15 +15,33 @@
 package template
 
 import (
+	"maps"
 	"strings"
+	"sync"
 	"text/template"
 
 	"cuelang.org/go/cue"
 )
 
+type FuncMap template.FuncMap
+
+var (
+	fm   = make(template.FuncMap)
+	muFM = sync.Mutex{}
+)
+
+// AddFuncMap copies funcMap provided by the user to global map.
+// If AddFuncMap is called multiple times with a key already present in global map,
+// the value in the global map will be overwritten by the latest value associated
+func AddFuncMap(funcMap FuncMap) {
+	muFM.Lock()
+	defer muFM.Unlock()
+	maps.Copy(fm, funcMap)
+}
+
 // Execute executes a Go-style template.
 func Execute(templ string, data cue.Value) (string, error) {
-	t, err := template.New("").Parse(templ)
+	t, err := template.New("").Funcs(fm).Parse(templ)
```

Comment:
This should be protected by the mutex

Otherwise, if Execute is called concurrently with AddFuncMap, there will be a race

Reply:
Thanks @verdverm. Does this work?
https://github.com/cue-lang/cue/blob/1c3c202cde93828510abf15c0498605a4dcc6d43/pkg/text/template/manual.go#L44-L46